src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/WalletController.cs

```diff
@@ -42,6 +63,100 @@ public async Task<ActionResult<ResultList<Claim>>> GetClaims([FromServices] IUni
             End = end != null ? DateTimeOffset.FromUnixTimeSeconds(end.Value) : null,
         });
 
-        return new ResultList<Claim> { Result = claims.Select(c => c.MapToV1()).ToArray() };
+        return new ResultList<Claim> { Result = claims.Select(c => c.MapToV1()) };
+    }
+
+    /// <summary>
+    /// Aggregates certificates based on the specified time zone and time range.
+    /// </summary>
+    /// <param name="unitOfWork"></param>
+    /// <param name="timeAggregate">The size of each bucket in the aggregation</param>
+    /// <param name="timeZone">The time zone. See https://en.wikipedia.org/wiki/List_of_tz_database_time_zones for a list of valid time zones.</param>
+    /// <param name="start">The start of the time range in Unix time.</param>
+    /// <param name="end">The end of the time range in Unix time.</param>
+    /// <param name="type">Filter the type of certificates to return.</param>
+    /// <response code="200">Returns the aggregated claims.</response>
+    /// <response code="401">If the user is not authenticated.</response>
+    [HttpGet]
+    [Route("v1/aggregate-certificates")]
+    [Produces("application/json")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
+    public async Task<ActionResult<ResultList<CertificateAggregationResult>>> AggregateCertificates(
+        [FromServices] IUnitOfWork unitOfWork,
+        [FromQuery] TimeAggregate timeAggregate,
+        [FromQuery] string timeZone, //https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+        [FromQuery] long? start,
+        [FromQuery] long? end,
+        [FromQuery] CertificateType? type)
```

Comment:
when i think of a restfull like api this would be the normal way to do it,

mywallet.com/v1/aggregate-certificates?start=13254&type=consumption   <-- return only consumption
mywallet.com/v1/aggregate-certificates?start=13254&type=production       <-- return only production
mywallet.com/v1/aggregate-certificates?start=13254                                       <-- filter not set, so return both types

otherwise i could use the same logic for start and end, why not require them? optional parameters makes developer expericences better since it removes the need to understand all parameters from the start.

Reply:
Okay, let's keep it. If we get negative feedback about this, then we change it in a later version.